apps/communications/contacts/test/marionette/import_sdcard_test.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+var Contacts = require('./lib/contacts');
+var assert = require('assert');
+
+marionette('Import from SDCARD', function() {
+  var profile = Contacts.config;
+  // Enable device storage
+  var prefs = ['device.storage.enabled',
+        'device.storage.testing',
+        'device.storage.prompt.testing'];
+  prefs.forEach(pref => {
+    profile[pref] = true;
+  });
+  var client = marionette.client({
+    profile: profile,
+    desiredCapabilities: { raisesAccessibilityExceptions: false }
+  });
+  var actions, subject, selectors;
+
+  setup(function() {
+    // Remove all files in temp device storage.
+    client.fileManager.removeAllFiles();
+
+    // Add the default vcard file
+    client.fileManager.add({
+      type: 'text',
+      filePath: 'apps/communications/contacts/test/marionette/data/vcard_4.vcf'
+    });
+
+    subject = new Contacts(client);
+    actions = client.loader.getActions();
+    subject.actions = actions;
+    subject.launch();
+    selectors = Contacts.Selectors;
+  });
+
+  test('> one contact in one file', function() {
+    // Go to settings
+    client.helper.waitForElement(selectors.settingsButton).click();
```

Comment:
We can make this test easier to read, if we implement the [Page Object pattern](https://wiki.mozilla.org/B2G/QA/Automation/Style_Guide/Best_Practices#The_Page_Object_Pattern). We could have something like:
```js
test('> one contact in one file', function() {
  var settings = contactsApp.goToSettings();
  var importContact = settings.goToImportContacts();
  var importContactFromSdCardModal = importContact.startImportFromSdCard()
  importContact = importContactFromSdCardModal.waitToBeOver();

  settings = importContact.goBack();
  contacts = settings.goBack();
  assert.deepStrictEqual(contactsApp.contacts, ['Forrest Gump']);
});
```
Having these kind of steps in the tests brings us closer to what an end-user intends to do.
Then, the implementation details (aka how a user would actually interact with the UI) should belong in the page class.
For example, in `contacts.js`:
```js
goToSettings: function() {
   this.client.helper.waitForElement(selectors.settingsButton).tap();
   return new Settings(this.client);
}
```

And then, you create another file, that is in charge of the "Settings page":
```js
function Settings(client) {
   this.client = client;
   this.client.helper.waitForElement(selectors.settingsHeader);
}
```

And so on.

Also, there is a transition when you tap on "import contacts". A good way to make sure we wait on it is the following:
```js
function ImportContacts(client) {
   this.client = client;
   var panel = this.client.helper.waitForElement(selectors.importContactsPanel);
   client.helper.waitFor(function() {
      return panel.rect().x == 0
   })
}
```